### PR TITLE
added option to hide ignore and later buttons

### PR DESCRIPTION
**Pull request by PonnamKarthik**
added option to hide ignore and later buttons
added option to close dialog on click on outside & added option for user to force by adding '<sparkle:criticalUpdate />' in tags

### CODE REVIEW

File: lib/src/upgrade_alert.dart [hunk 1]
```diff
@@ -6,6 +6,8 @@ import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 import 'package:http/http.dart' as http;
 
+import '../upgrader.dart';
+import 'upgrader.dart';
```
Comment (larryaasen) [inline lib/src/upgrade_alert.dart]: @PonnamKarthik Thanks for the PR. The travis-ci build of this branch is failing, in one case for this duplicate import. Perhaps this is due to a merge problem?
https://travis-ci.org/larryaasen/upgrader/pull_requests

**Comment by PonnamKarthik** [comment]
@larryaasen updated the Code

File: lib/src/upgrade_alert.dart (hunk 1 above)
Comment (PonnamKarthik) [reply lib/src/upgrade_alert.dart]: @larryaasen updated the Code

**Comment by PonnamKarthik** [comment]
@larryaasen updated the Code

**Comment by PonnamKarthik** [comment]
@larryaasen lowered the flutter support to 1.12.0 from 1.13.0

File: lib/src/upgrade_alert.dart [hunk 2]
```diff
@@ -209,24 +235,26 @@ class UpgradeCard extends _UpgradeBase {
...
-                        FlatButton(
-                            child: Text(Upgrader().buttonTitleLater),
-                            onPressed: () {
-                              // Save the date/time as the last time alerted.
-                              Upgrader().saveLastAlerted();
-
-                              Upgrader().onUserLater(context, false);
-                              state.forceUpdateState();
-                            }),
+                        if (Upgrader().showIgnore)
```
[the remark this answers is not in the document]
Comment (PonnamKarthik) [reply lib/src/upgrade_alert.dart]: @larryaasen we cannot use `{` here

File: lib/src/upgrade_alert.dart [hunk 3]
```diff
@@ -209,24 +235,26 @@ class UpgradeCard extends _UpgradeBase {
...
+                          FlatButton(
+                              child: Text(Upgrader().buttonTitleIgnore),
+                              onPressed: () {
+                                // Save the date/time as the last time alerted.
+                                Upgrader().saveLastAlerted();
+
+                                Upgrader().onUserIgnored(context, false);
+                                state.forceUpdateState();
+                              }),
+                        if (Upgrader().showLater)
```
[the remark this answers is not in the document]
Comment (PonnamKarthik) [reply lib/src/upgrade_alert.dart]: @larryaasen we cannot use `{` here

File: lib/src/upgrade_alert.dart [hunk 4]
```diff
@@ -209,24 +235,26 @@ class UpgradeCard extends _UpgradeBase {
...
+                          FlatButton(
+                              child: Text(Upgrader().buttonTitleLater),
+                              onPressed: () {
+                                // Save the date/time as the last time alerted.
+                                Upgrader().saveLastAlerted();
+
+                                Upgrader().onUserLater(context, false);
+                                state.forceUpdateState();
+                              }),
```
[the remark this answers is not in the document]
Comment (PonnamKarthik) [reply lib/src/upgrade_alert.dart]: @larryaasen we cannot use `{` here

File: lib/src/upgrader.dart [hunk 5]
```diff
@@ -319,12 +339,14 @@ class Upgrader {
...
-                child: Text(buttonTitleLater),
-                onPressed: () => onUserLater(context, true)),
+            if (showIgnore)
+              FlatButton(
+                  child: Text(buttonTitleIgnore),
+                  onPressed: () => onUserIgnored(context, true)),
+            if (showLater)
+              FlatButton(
+                  child: Text(buttonTitleLater),
+                  onPressed: () => onUserLater(context, true)),
```
[the remark this answers is not in the document]
Comment (PonnamKarthik) [reply lib/src/upgrader.dart]: @larryaasen we cannot use { here

File: lib/src/upgrader.dart [hunk 6]
```diff
@@ -319,12 +339,14 @@ class Upgrader {
...
-                child: Text(buttonTitleIgnore),
-                onPressed: () => onUserIgnored(context, true)),
-            FlatButton(
-                child: Text(buttonTitleLater),
-                onPressed: () => onUserLater(context, true)),
+            if (showIgnore)
+              FlatButton(
+                  child: Text(buttonTitleIgnore),
+                  onPressed: () => onUserIgnored(context, true)),
+            if (showLater)
```
[the remark this answers is not in the document]
Comment (PonnamKarthik) [reply lib/src/upgrader.dart]: @larryaasen we cannot use { here

**Pull request merged**